src/proc/typifier.rs

```diff
@@ -704,6 +704,7 @@ impl<'a> ResolveContext<'a> {
                     Mf::Min |
                     Mf::Max |
                     Mf::Clamp |
+                    Mf::Saturate |
```

Comment:
This seems like the wrong place to put it, if we look at the code for this match arm we have:

```rust
match *res_arg.inner_with(types) {
  Ti::Vector {
      kind,
      size: _,
      width,
  } => TypeResolution::Value(Ti::Scalar { kind, width }),
  ref other =>
      return Err(ResolveError::IncompatibleOperands(
          format!("{:?}({:?}, _)", fun, other)
      )),
},
```

but saturate is defined as

```
@const fn saturate(e: T) -> T
where
S is AbstractFloat, f32, or f16
T is S or vecN<S>

Reply:
Thanks, making the changes now, this one appears to end on `Mf::Pow => res_arg.clone(),`

Reply:
Oh yeah sorry confused with the match body for `Dot`